tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -114,21 +114,13 @@ public static Optional<TraceMetadata> maybeGetTraceMetadata() {
             return Optional.empty();
         }
 
-        TraceMetadata.Builder builder = TraceMetadata.builder().traceId(trace.getTraceId());
-        String requestId = trace.maybeGetRequestId();
-        if (requestId != null) {
-            builder.requestId(requestId);
-        }
+        OpenSpan span = trace.top().orElse(null);
 
-        if (trace.isObservable()) {
-            return trace.top().map(openSpan -> builder.spanId(openSpan.getSpanId())
-                    .parentSpanId(openSpan.getParentSpanId())
-                    .build());
-        } else {
-            return Optional.of(builder.spanId(Tracers.randomId())
-                    .parentSpanId(Optional.empty())
-                    .build());
-        }
```

Comment:
The current behavior is inconsistent:
- If the current trace is sampled with no span, then we return an absent value.
- If the current trace is unsampled (with no span), then we return a present value with a made-up span ID.

We should always propagate trace information if we have it. It's unfortunate that `TraceMetadata.spanId` is a required field - really this should be optional.